src/it/scala/io/iohk/ethereum/sync/util/SyncUtils.scala

```diff
@@ -0,0 +1,78 @@
+package io.iohk.ethereum.sync.util
+
+import java.net.{InetSocketAddress, ServerSocket}
+import java.util.concurrent.TimeoutException
+
+import akka.util.ByteString
+import io.iohk.ethereum.Mocks.MockValidatorsAlwaysSucceed
+import io.iohk.ethereum.domain.{Account, Address, Block, Receipt}
+import io.iohk.ethereum.ledger.{BlockExecutionError, BlockExecutionSuccess, InMemoryWorldStateProxy}
+import io.iohk.ethereum.nodebuilder.ShutdownHookBuilder
+import monix.eval.Task
+import io.iohk.ethereum.utils.Logger
+
+import scala.concurrent.duration.FiniteDuration
+
+object SyncUtils {
+
+  object ShutdownHookBuilder extends ShutdownHookBuilder with Logger
+  class ValidatorsExecutorAlwaysSucceed extends MockValidatorsAlwaysSucceed {
+    override def validateBlockAfterExecution(block: Block, stateRootHash: ByteString, receipts: Seq[Receipt], gasUsed: BigInt)
+    : Either[BlockExecutionError, BlockExecutionSuccess] = Right(BlockExecutionSuccess)
+  }
+
+  object ValidatorsExecutorAlwaysSucceed extends ValidatorsExecutorAlwaysSucceed
+
+  def retryUntilWithDelay[A](source: Task[A], delay: FiniteDuration, maxRetries: Int)(
+    predicate: A => Boolean
+  ): Task[A] = {
+    source.delayExecution(delay).flatMap { result =>
+      if (predicate(result)) {
+        Task.now(result)
+      } else {
+        if (maxRetries > 0) {
+          retryUntilWithDelay(source, delay, maxRetries - 1)(predicate)
+        } else {
+          Task.raiseError(new TimeoutException("Task time out after all retries"))
+        }
+      }
+    }
+  }
```

Comment:
Maybe we would like to have tests for `retryUntilWithDelay` for example:
* fail when task success only on maxRetries + 1 attempts
* success when task succeed on maxRetries attempt
* what happens when task fails to be clear about this case